src/models/connection.rs

```diff
@@ -92,3 +115,103 @@ pub enum SortBy {
     #[graphql(name = "RECENTLY_PLAYED")]
     RecentlyPlayed,
 }
+
+pub trait GetConnection<TB>
+where
+    Self: HasTable<Table = TB> + Queryable<TB::SqlType, Sqlite> + Sized,
+    TB: Table
+        + BoxedDsl<
+            'static,
+            Sqlite,
+            Output = BoxedSelectStatement<'static, <TB as AsQuery>::SqlType, TB, Sqlite>,
+        >,
+    <TB as QuerySource>::FromClause: QueryFragment<Sqlite>,
+    Sqlite: HasSqlType<TB::SqlType>,
+{
+    type Name: Column<Table = TB, SqlType = Text>
+        + AppearsOnTable<TB>
+        + QueryFragment<Sqlite>
+        + NonAggregate;
+    type TimeAdded: Column<Table = TB, SqlType = Timestamp>
+        + AppearsOnTable<TB>
+        + QueryFragment<Sqlite>;
+    type LastPlayed: Column<Table = TB, SqlType = Nullable<Timestamp>>
+        + AppearsOnTable<TB>
+        + QueryFragment<Sqlite>;
+
+    fn name() -> Self::Name;
+    fn time_added() -> Self::TimeAdded;
+    fn last_played() -> Self::LastPlayed;
+
+    fn get_connection(
+        context: &GraphQLContext,
+        first: i64,
+        after: Option<String>,
+        sort: Option<SortParams>,
+    ) -> FieldResult<Connection<Self>> {
```

Comment:
Return a `diesel::QueryResult`. Converting to `FieldResult` here is basically casting the error into a string.

Reply:
Hm, I'd like for `FieldResult` to only be used in the `graphql_object` macro but since this function is (and really only can be) used in the context of a graphql response, it isn't that big a deal.